Editor/Window/DeploymentScenariosInput.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using Editor.Window;
+using UnityEngine;
+using UnityEngine.UIElements;
+
+namespace AmazonGameLift.Editor
+{
+    internal class DeploymentScenariosInput : StatefulInput
+    {
+        private InputState _inputState;
+        private bool _enabled;
+        private DeploymentScenarios _deploymentScenarios;
+
+        private readonly VisualElement _container;
+        private readonly VisualElement _radio2Group;
+        private readonly VisualElement _radio3Group;
+        private readonly Button _showMoreScenariosButton;
+
+        public Action<DeploymentScenarios> OnValueChanged;
+
+        public static readonly Dictionary<int, DeploymentScenarios> ScenarioIndexMap = new()
+        {
+            { 1, DeploymentScenarios.SingleRegion },
+            { 3, DeploymentScenarios.SpotFleet },
+            { 4, DeploymentScenarios.FlexMatch },
+        };
+
+        public DeploymentScenariosInput(VisualElement container, DeploymentScenarios initialValue, bool enabled)
+        {
+            var uxml = Resources.Load<VisualTreeAsset>("EditorWindow/Components/DeploymentScenarios");
+            container.Add(uxml.Instantiate());
+
+            _container = container;
+            _inputState = initialValue == DeploymentScenarios.SingleRegion ? InputState.Initial : InputState.Expanded;
+            _deploymentScenarios = initialValue;
+            _enabled = enabled;
+            _container.SetEnabled(_enabled);
+
+            _radio2Group = container.Q("FleetTypeRadioButton2Group");
+            _radio3Group = container.Q("FleetTypeRadioButton3Group");
```

Comment:
Please rename the elements to be relevant to the deployment scenario instead of using a numbering based convention.

Reply:
renamed